### PR TITLE
TICK-385 Added fast-math and O3 compilation flags

### PR DESCRIPTION
**Pull request by vinther**

### CODE REVIEW

**Comment by stephanegaiffas** [comment]
I have the same error as Travis's on my laptop. 
Seems to be compiling fine, but C++ tests seem to fail

**Comment by vinther** [comment]
Some silly mistake prevented the build from passing. Trying again now.

**Comment by stephanegaiffas** [comment]
There is still a problem (I just pulled to get last version of the branch) :
CMake Error at CMakeLists.txt:83 (add_subdirectory):
  add_subdirectory given source "benchmark" which is not an existing
  directory.

**Comment by vinther** [comment]
I think something is wrong on your end

File: setup.py [hunk 1]
```diff
@@ -37,6 +37,10 @@
 
 debug_flags = ['DEBUG_COSTLY_THROW']
 
+# If true, add compilation flags to use fast (but maybe inaccurate) math
+# See https://gcc.gnu.org/wiki/FloatingPointMath
+use_fast_math = False
```
Comment (Mbompr) [inline setup.py]: You've convinced me that default should be True

**Comment by Mbompr** [comment]
@stephanegaiffas Everything is fine on my machine

**Comment by stephanegaiffas** [comment]
Okay problem solved

**Comment by stephanegaiffas** [comment]
Don't merge ! Still not happy with : some python unit tests are failing (which is perfectly normal)
I'll correct

**Pull request merged**